File: crime-analytics-backend/src/main/java/com/crimeanalytics/service/ReportService.java
package com.crimeanalytics.service;

import com.crimeanalytics.dto.request.ReportRequest;
import com.crimeanalytics.dto.request.response.ReportResponse;

import java.util.List;

public interface ReportService {

    ReportResponse createReport(ReportRequest request);

    List<ReportResponse> getAllReports();

    ReportResponse getReportById(Long id);

    List<ReportResponse> getReportsByType(String reportType);

    List<ReportResponse> getReportsByGeneratedBy(String generatedBy);

    ReportResponse updateReport(Long id, ReportRequest request);

    void deleteReport(Long id);
}

File: crime-analytics-frontend/src/services/reportService.js
import axiosInstance from '../api/axiosInstance';

export const reportService = {
  async getAllReports() {
    const response = await axiosInstance.get('/reports');
    return response.data;
  },

  async getReportById(id) {
    const response = await axiosInstance.get(`/reports/${id}`);
    return response.data;
  },

  async createReport(data) {
    // data: { reportName, reportType, generatedBy, generatedAt, filePath, description }
    const response = await axiosInstance.post('/reports', data);
    return response.data;
  },

  async getByType(reportType) {
    const response = await axiosInstance.get(`/reports/type/${encodeURIComponent(reportType)}`);
    return response.data;
  },

  async getByGeneratedBy(generatedBy) {
    const response = await axiosInstance.get(`/reports/generated-by/${encodeURIComponent(generatedBy)}`);
    return response.data;
  },

  async updateReport(id, data) {
    const response = await axiosInstance.put(`/reports/${id}`, data);
    return response.data;
  },

  async deleteReport(id) {
    const response = await axiosInstance.delete(`/reports/${id}`);
    return response.data;
  }
};
